### DISCUSSION TITLE
Wazuh Windows agent gaping security hole!

### DISCUSSION THREAD

**Discussion by jakko10** [Security-Onion-Solutions/securityonion#9390] [General]
Hi!

Looks like Security Onion still uses the old Wazuh agent version 3.13.1-1 that has an (in my personal opinion) critical level security hole in it.
The ossec.conf file permissions are incorrect. The Everyone user group has read and write permissions. The latter is the critical one.
Due to this bug, any user who is not an administrator could easily elevate to administrator/system asses rights.
Why and how:
The why: Wazuh agent service rus as local system account.
The how: If one adds a script wodle to Wazuh agent conf (ossec.conf) the script will run in the system context and can do pretty much anything. After the next reboot, the script gets loaded and magic happens.

**Comment by dougburks** [comment]
Hi @jakko10,

We've tested this on a few different machines and here are our findings.

We downloaded the Wazuh agent:
![Screenshot 2022-12-14 at 1 05 10 PM](https://user-images.githubusercontent.com/1659467/207682153-01da0fa2-3be5-47bd-8a22-b1938b8d19fa.png)


and verified the hash:
![Screenshot 2022-12-14 at 1 08 55 PM](https://user-images.githubusercontent.com/1659467/207682157-b1a11256-97bb-4c29-9c66-51d66cf7e9b6.png)


We then installed the agent and verified that the permissions on ossec.conf allowed all users to write:
![Screenshot 2022-12-14 at 1 46 20 PM](https://user-images.githubusercontent.com/1659467/207685392-ed1b207d-5cb3-41c3-ab20-b48219b7c5be.png)


We then ran the Wazuh Agent Manager:
![Screenshot 2022-12-14 at 1 41 18 PM](https://user-images.githubusercontent.com/1659467/207684240-5dc3bd06-723b-47e4-8774-d4d004b58180.png)


and configured the agent to connect to the Wazuh server. Saving the configuration locks down the permissions on ossec.conf:
![Screenshot 2022-12-14 at 1 35 34 PM](https://user-images.githubusercontent.com/1659467/207682161-a205f4bc-241a-44e4-9b2d-a08fa9991b8f.png)


Can you confirm that saving the configuration locks down the permissions on ossec.conf for you?

**Reply by jakko10** [reply]
Aha ok, so, yes, different installation methods produce different results.
If you deploy the Wazuh agent in scale let's say with Ansible (Wazuh has written the Ansible roles to automate the agent registration) then even if the agent is registered and all that, the permissions still stay so that "Everyone" users group is still present with the read/write permissions.
There are multiple ways to install it programmatically, producing the same result leaving ossec.conf vulnerable.
I did report this a while back to Wazuh too, they have fixed this in the new versions so that ossec.conf always has the correct permissions.

**Reply by dougburks** [reply]
I've added the following warning to https://docs.securityonion.net/en/2.3/wazuh.html#adding-agents:
![image](https://user-images.githubusercontent.com/1659467/207693040-d9e87505-d49d-4650-b16b-7206837cdd80.png)

In the future, if you need to report security issues to us, please follow our Vulnerability Disclosure steps at:
https://docs.securityonion.net/en/2.3/security.html#vulnerability-disclosure

Thanks!

**Reply by jakko10** [reply]
I actually did alert you guys regarding this issue about 4 years ago... yep that hole has been there for this long. I alerted Wazuh too back then and ended up testing the agent with them for quite some time. Probably should have created a CVE.
I guess the note is a good start.
However, I don't think most users would configure Wazuh using Wazuh Agent Manager, I mean, small companies with 10-20 PC's probably would, but as you scale up. You really don't want to go from PC to PC manually configuring things, you'd want to do things remotely and automatically.

**Reply by dougburks** [reply]
I've also published a blog post:
https://blog.securityonion.net/2022/12/potential-security-issue-in-windows.html
